### PR TITLE
Pypyr go terra for building pipelines with Terraform

### PR DESCRIPTION
**Pull request by AgrawalRahul**

### CODE REVIEW

File: pypyr-go-terra/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+RUN apt-get update && apt-get install -y unzip
+
+# unzip the Terraform zip file
+RUN unzip terraform_0.11.8_linux_amd64.zip
+
+# move Terraform executable to $PATH
+RUN mv terraform /usr/local/bin
+
+# remove the downloaded Terraform .zip file
+RUN rm -f terraform_0.11.8_linux_amd64.zip
```
Comment (345thomas) [inline pypyr-go-terra/Dockerfile]: line 12-24 should all be a single RUN instruction, in order to minimize layers in the docker image.  i.e
```
RUN wget https://releases.hashicorp.com/terraform/0.11.8/terraform_0.11.8_linux_amd64.zip; \
         apt-get update && apt-get install -y unzip; \
         unzip terraform_0.11.8_linux_amd64.zip; \
         mv terraform /usr/local/bin; \
         rm -f terraform_0.11.8_linux_amd64.zip
```

File: pypyr-go-terra/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# be aware that the WORKDIR is NOT the GOPATH.
+# the WORKDIR is /src, which is the pypyr working dir. this will let you mount a
+# volume with a pipelines dir in it without hiding the go path.
+# build me like this from repo root:
+# docker build -t pypyr/pypyr-go-terra -t pypyr/pypyr-go-terra:latest -t pypyr/pypyr-go-terra:0.11 -t pypyr/pypyr-go-terra:0.11.8 -f pypyr-go-terra/Dockerfile .
+
+FROM pypyr/pypyr-go:pypyr1.0.0-go1.11
+
+# get the Terraform executable as zip
+RUN wget https://releases.hashicorp.com/terraform/0.11.8/terraform_0.11.8_linux_amd64.zip
```
Comment (345thomas) [inline pypyr-go-terra/Dockerfile]: if you declare `ARG terraformversion=0.11.8` on top after the FROM, then you only need to update it in one place when bumping versions. . .

File: ops/deploy.sh [hunk 3]
```diff
@@ -5,7 +5,7 @@
 # Usage
 # From repo:
 # ./ops/deploy.sh [deployprefix]
-# where deployprefix is the name of one of the image subdirs, e.g pypyr or pypyr-go
+# where deployprefix is the name of one of the image subdirs, e.g pypyr or pypyr-go or pypyr-go-terra
```
Comment (345thomas) [inline ops/deploy.sh]: break line before `pypyr-go`

File: pypyr-go-terra/README.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+$ docker run -v ${PWD}:/src pypyr/pypyr-go-terra simple
+
+# and this is how you pass further parameters like context
+# this command looks for ./pipelines/substitutions.yaml, which is on your host.
+$ docker run -v ${PWD}:/src pypyr/pypyr-go-terra substitutions "key1=this is key1 in context,key2=pipe"
+
+# one more time, with extra logging. . .
+$ docker run -v ${PWD}:/src pypyr/pypyr-go-terra substitutions "key1=this is key1 in context,key2=pipe" --log 10
+```
+
```
Comment (345thomas) [inline pypyr-go-terra/README.md]: pls add something like
### using terraform in a pipeline

You can use terraform in a pipeline like this:
```steps:
             name: pypyr.steps.cmd
              in:
                  cmd: terraform blah blah
```

Maybe something more useful than blah blah :smile:

**Comment by AgrawalRahul** [comment]
@345thomas Please check now.

File: pypyr-go-terra/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+# the WORKDIR is /src, which is the pypyr working dir. this will let you mount a
+# volume with a pipelines dir in it without hiding the go path.
+# build me like this from repo root:
+# docker build -t pypyr/pypyr-go-terraform -t pypyr/pypyr-go-terraform:latest -t pypyr/pypyr-go-terraform:0.11 -t pypyr/pypyr-go-terraform:0.11.8 -f pypyr-go-terraform/Dockerfile .
+
+FROM pypyr/pypyr-go:pypyr1.0.0-go1.11
+
+ARG terraformversion=0.11.8
+
+# get the Terraform executable as zip
```
Comment (345thomas) [inline pypyr-go-terra/Dockerfile]: not a useful comment - it just describes what the 1st line wget does. think the code is clear enough you can just get rid of comment,

**Comment by 345thomas** [comment]
test

**Pull request merged**